hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,56 @@
+MONGO_DB_NAME = pulp_db_param.value
+
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def migration
+  export_dir = '/var/tmp/engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+  mongo_backup = '/var/tmp/mongo_backup'
+  mongo_conf = '/etc/opt/rh/rh-mongodb34/mongod.conf'
+
+  # Verify we have a good value in the answerfile for the MongoDB DB name.
+  pulp_db_param = param('katello', 'pulp_db_name') || param('foreman_proxy_content', 'pulp_db_name')
+  fail_and_exit('Unknown pulp database name') unless pulp_db_param
+
+  # Create export directory and dump MongoDB
+  logger.info 'Stopping Pulp services except MongoDB'
+  Kafo::Helpers.execute('systemctl stop rh-mongodb34-mongod')
+  FileUtils.mkdir_p(export_dir)
+  File.chmod(0700, export_dir)
+  logger.info "Starting mongodump to #{export_dir}"
+  Kafo::Helpers.execute("mongodump --host localhost --out #{export_dir}")
```

Comment:
This errors for me if the database is stopped by line 21

Reply:
Ahh, cause you went with systemctl. This is why I prefer we just use `katello-service` so we don;t get into sometimes this and sometimes that. In the case of line 21 that needs to be stopping all services BUT mongodb and thus using katello-service.